### ISSUE TITLE
Error on rollback

### ISSUE THREAD

**Issue by bkmo** [adonespitogo/node-snapback#3]
I am getting this on rollback with efi part mounted to /boot/efi and a boot partition mounted to /boot:
`Removing contents of /boot...
Error in rollback:  [Error: EBUSY: resource busy or locked, rmdir '/boot/efi'] {
  errno: -16,
  code: 'EBUSY',
  syscall: 'rmdir',
  path: '/boot/efi'
System was rendered not bootable, dropped to grub shell.`

I am going to do an install with an encrypted /boot and see how it goes without the unencrypted /boot partition.
Same error with encrypted boot. It is something with node-snapback because snapper-rollback is functioning properly.

**Comment by adonespitogo** [comment]
This does not currently work with if your ESP is not the /boot directory.

**Comment by adonespitogo** [comment]
This is the [current implementation](https://github.com/adonespitogo/node-snapback/blob/main/actions/rollback.js) of `rollback` command, perhaps you can try to debug it on your end.

**Comment by bkmo** [comment]
I have 2 partitions one is the EFI mounted to /boot/efi and the other is boot mounted to /boot. So there seems to be an issue with the way it is handling /boot/efi.

**Comment by adonespitogo** [comment]
Yes definitely there's an issue. The script is trying to `rm -rf /boot/*` during rollback. But your `/boot` contains mounted partition which cannot be removed, hence the error.

**Comment by bkmo** [comment]
I see where it would work as designed if you have a "Pure" arch install with the /boot residing on the same partition as the ESP. I am testing on a VM with a calamres install that is setup to install the bootloader (grub) at /boot/efi

**Comment by adonespitogo** [comment]
Hi, I think by not removing the contents of /boot, it will work with your setup. Can you try to update to `v1.1.1` and see if works?

**Comment by bkmo** [comment]
I will work on it later this evening, or tomorrow.

**Comment by bkmo** [comment]
version 1.1.1 did not give me an error and performed the rollback. I was thinking this tool might make it easy to boot from snapshots without having kernel module mismatches after a kernel update. But this is not the case. Only the backup that is created from the tool is going to boot properly after a kernel update. I will just stick with keeping a locked version of kernel install to boot from snapshots.

**Comment by adonespitogo** [comment]
If you can find a way to hook the backup script to your work flow, then you will be able to rollback successfully. The backup script is located at `/usr/local/bin/bootbackup.sh`.

**Comment by adonespitogo** [comment]
I've added more targets to the hook as suggested in https://github.com/adonespitogo/node-snapback/issues/2

If you have the chance, try to update to version `1.1.2` and do `sudo snapback install` again to install the new pacman hook.
